### PR TITLE
sanitizer: sanitize vin number

### PR DESCRIPTION
**Pull request by jnewb1**

### CODE REVIEW

File: selfdrive/test/helpers.py [hunk 1]
```diff
@@ -82,7 +82,20 @@ def read_segment_list(segment_list_path):
 
 # Utilities for sanitizing routes of only essential data for testing car ports and doing validation.
 
+def sanitize_vin(vin: str): # remove sentitive data from VIN
+  VIN_SENSITIVE = 6 # (last 6 digits of vin are serial number https://en.wikipedia.org/wiki/Vehicle_identification_number)
+  return vin[:-VIN_SENSITIVE] + "X" * VIN_SENSITIVE
+
```
Comment (sshane) [inline selfdrive/test/helpers.py:88]:
```suggestion


```

File: selfdrive/test/helpers.py [hunk 2]
```diff
@@ -82,7 +82,20 @@ def read_segment_list(segment_list_path):
 
 # Utilities for sanitizing routes of only essential data for testing car ports and doing validation.
 
+def sanitize_vin(vin: str): # remove sentitive data from VIN
+  VIN_SENSITIVE = 6 # (last 6 digits of vin are serial number https://en.wikipedia.org/wiki/Vehicle_identification_number)
```
Comment (sshane) [inline selfdrive/test/helpers.py:86]:
```suggestion
def sanitize_vin(vin: str):  # remove sentitive data from VIN
  VIN_SENSITIVE = 6  # (last 6 digits of vin are serial number https://en.wikipedia.org/wiki/Vehicle_identification_number)
```

for readability/PEP: https://peps.python.org/pep-0008/#inline-comments

also probably doesn't need to be inline at all

File: selfdrive/test/helpers.py [hunk 3]
```diff
@@ -82,7 +82,20 @@ def read_segment_list(segment_list_path):
...
+  return vin[:-VIN_SENSITIVE] + "X" * VIN_SENSITIVE
+
+def sanitize_msg(msg: LogMessage) -> LogMessage:
+  if msg.which() == "carParams":
+    msg = msg.as_builder()
+    msg.carParams.carVin = sanitize_vin(msg.carParams.carVin)
+    msg = msg.as_reader()
+  return msg
+
 PRESERVE_SERVICES = ["can", "carParams", "pandaStates", "pandaStateDEPRECATED"]
```
Comment (sshane) [inline selfdrive/test/helpers.py:97]:
```suggestion


```

File: selfdrive/test/helpers.py [hunk 4]
```diff
@@ -82,7 +82,20 @@ def read_segment_list(segment_list_path):
...
+  VIN_SENSITIVE = 6 # (last 6 digits of vin are serial number https://en.wikipedia.org/wiki/Vehicle_identification_number)
+  return vin[:-VIN_SENSITIVE] + "X" * VIN_SENSITIVE
+
+def sanitize_msg(msg: LogMessage) -> LogMessage:
+  if msg.which() == "carParams":
+    msg = msg.as_builder()
+    msg.carParams.carVin = sanitize_vin(msg.carParams.carVin)
+    msg = msg.as_reader()
+  return msg
+
```
Comment (sshane) [inline selfdrive/test/helpers.py:95]:
```suggestion


```

File: selfdrive/test/helpers.py (hunk 2 above)
Comment (jnewb1) [reply selfdrive/test/helpers.py:86]: any ruff rules for these?

**Comment by Notstupe** [comment]
Are you sure you didn't sanitize right?

File: selfdrive/test/helpers.py (hunk 2 above)
Comment (sshane) [reply selfdrive/test/helpers.py]: Not yet

**Pull request merged**